src/Nethermind/Nethermind.JsonRpc/Modules/Eth/EthRpcModule.TransactionExecutor.cs

```diff
@@ -109,12 +110,23 @@ public ResultWrapper<TResult> ExecuteTx(TransactionForRpc transactionCall, Block
 
             protected abstract ResultWrapper<TResult> ExecuteTx(BlockHeader header, Transaction tx, Dictionary<Address, AccountOverride>? stateOverride, CancellationToken token);
 
-            protected ResultWrapper<TResult> CreateResultWrapper(bool inputError, string? errorMessage, TResult? bodyData)
+            protected ResultWrapper<TResult> CreateResultWrapper(bool inputError, string? errorMessage, TResult? bodyData, bool executionReverted)
             {
-                if (inputError)
-                    return ResultWrapper<TResult>.Fail(errorMessage, ErrorCodes.InvalidInput);
-                if (errorMessage is not null)
+                if (inputError || errorMessage is not null)
+                {
+                    if (executionReverted)
+                    {
+                        if (bodyData is not null)
+                        {
+                            return ResultWrapper<TResult>.Fail("execution reverted: " + errorMessage, ErrorCodes.ExecutionReverted, bodyData);
+                        }
+
+                        var errorData = errorMessage is not null ? Encoding.UTF8.GetBytes(errorMessage).ToHexString(true) : null;
```

Comment:
So Revert returns byte array. And we do this converting into string and back. Maybe we should pass this data through `CallOutput.OutputData`. I think it already there